runhouse/servers/http/http_server.py

```diff
@@ -66,50 +68,47 @@ async def wrapper(*args, **kwargs):
         den_auth_enabled: bool = HTTPServer.get_den_auth()
         is_coro = inspect.iscoroutinefunction(func)
 
-        func_call: bool = func.__name__ in ["call_module_method", "call", "get_call"]
+        func_call: bool = func.__name__ in ["post_call", "get_call"]
         token = get_token_from_request(request)
 
-        if func_call and token:
-            obj_store.add_user_to_auth_cache(token, refresh_cache=False)
-
-        if not den_auth_enabled or func_call:
-            # If this is a func call, we'll handle the auth in the object store
-            if is_coro:
-                return await func(*args, **kwargs)
-
-            return func(*args, **kwargs)
-
-        if token is None:
-            raise HTTPException(
-                status_code=404,
-                detail="No token found in request auth headers. Expected in "
-                f"format: {json.dumps({'Authorization': 'Bearer <token>'})}",
-            )
+        request_id = request.headers.get("X-Request-ID", str(uuid.uuid4()))
+        token_hash = hash_token(token) if den_auth_enabled and token else None
+        _ctx_token = obj_store._set_ctx(request_id=request_id, token_hash=token_hash)
```

Comment:
also `._set_ctx` and `._get_ctx` should not be prefixed with underscores since they're called outside the obj store